FirebaseAuth/Sources/Public/FirebaseAuth/FIRTOTPMultiFactorGenerator.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import "FIRTOTPMultiFactorAssertion.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** @class FIRTOTPMultiFactorGenerator
+    @brief The data structure used to help initialize an assertion for a second factor entity to the
+        Firebase Auth/CICP server. Depending on the type of second factor, this will help generate
+        the assertion.
+        This class is available on iOS only.
+*/
+NS_SWIFT_NAME(TOTPMultiFactorGenerator) API_UNAVAILABLE(macos, tvos, watchos)
+    @interface FIRTOTPMultiFactorGenerator : NSObject
+
+/** @fn assertionForSignInWithenrollmentID:
+    @brief Initializes the MFA assertion to confirm ownership of the totp second factor. This
+   assertion is used to complete signIn with TOTP as a second factor.
+    @param enrollmentID The id that identifies the enrolled TOTP second factor.
+    @param oneTimePassword one time password string.
+*/
++ (FIRTOTPMultiFactorAssertion *)assertionForSignInWithEnrollmentID:(NSString *)enrollmentID
```

Comment:
Just double checking - since we had to amend `assertionForEnrollmentWithSecret` with a `completion` param,  do you think may need something similar here?

Reply:
No, this is a sync method.